booster-app/src/main/java/de/neuefische/boosterapp/service/BoosterService.java

```diff
@@ -31,10 +34,18 @@ public void deleteBooster(String id) {
         boosterDb.deleteById(id);
     }
 
-//    public Booster getRandomBooster(BoosterType randomBoost, String ownerId) {
-//        boosterDb.findAll()
-//
-//    }
+    public Booster getRandomBooster(BoosterType randomBoost, String owner) {
+        List<Booster> list = new ArrayList<>();
+        List<Booster> response = boosterDb.findByOwnerAndType(owner, randomBoost);
```

Comment:
findByOwnerAndType should return only booster with type and owner, therefor you don't need the for loop